test/collector.test.ts

```diff
@@ -0,0 +1,351 @@
+import { expect, use } from 'chai';
+import Collector from '../build/contracts/Collector.json';
+import TestToken from '../build/contracts/TestToken.json';
+import {
+    deployContract,
+    loadFixture,
+    MockProvider,
+    solidity
+} from 'ethereum-waffle';
+import { Wallet } from '@ethersproject/wallet';
+
+use(solidity);
+
+const FIRST_PARTNER_SHARE = 20;
+const SECOND_PARTNER_SHARE = 30;
+const THIRD_PARTNER_SHARE = 40;
+const FOURTH_PARTNER_SHARE = 10;
+
+const buildPartners = (partners: Wallet[], shares?: number[]) => {
+    return [
+        {
+            beneficiary: partners[0].getAddress(),
+            share: shares?.[0] ?? FIRST_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[1].getAddress(),
+            share: shares?.[1] ?? SECOND_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[2].getAddress(),
+            share: shares?.[2] ?? THIRD_PARTNER_SHARE
+        },
+        {
+            beneficiary: partners[3].getAddress(),
+            share: shares?.[3] ?? FOURTH_PARTNER_SHARE
+        }
+    ];
+};
+
+async function prepareAll(wallets: Wallet[]) {
+    const [
+        owner,
+        remainder,
+        partner1,
+        partner2,
+        partner3,
+        partner4,
+        utilWallet
+    ] = wallets;
+
+    const testToken = await deployContract(owner, TestToken);
+    const partners = buildPartners([partner1, partner2, partner3, partner4]);
+
+    const collector = await deployContract(owner, Collector, [
+        owner.address,
+        testToken.address,
+        partners,
+        remainder.address
+    ]);
+
+    return { collector, testToken, partners, remainder, utilWallet };
+}
+
+describe('Collector', () => {
+    async function prepareAllFixture(wallets: Wallet[]) {
+        return prepareAll(wallets);
+    }
```

Comment:
why does this have to be repeated here?

Reply:
My intention was to keep the logic related to preparation outside of the describe. The fixture should be part of the describe so this is the way to connect both.

Reply:
this is absolutely unnecessary. you can use the function directly, even though it is outside of the describe body.

Reply:
You're right on this. Thank you. Done.